notebooks/Gradient Boosted Regression Trees - Single Target.ipynb

```diff
@@ -0,0 +1,646 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "n_estimators=1000\n",
+    "clf_Ca=skl.ensemble.GradientBoostingRegressor(n_estimators=n_estimators)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "start=time()\n",
+    "clf_Ca.fit(X_train,y_train.values.ravel())\n",
+    "print(\"GBR took %.2f seconds for %d estimators\"\n",
+    "      % ((time() - start), n_estimators))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "y_pred = clf_Ca.predict(X_test)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here are some scoring metrics for our first pass at this model:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "skl.metrics.mean_squared_error(y_test,y_pred)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "skl.metrics.r2_score(y_test,y_pred)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Feature importances and selection"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true
+   },
+   "source": [
+    "There are ways to save time without giving up model performance.  One way is by using GradientBoostingRegressor's **feature_importances** attribute, which can identify the most important features in determining a split, allowing us to cast out less important features."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Below we print out the highest ranking features\n",
+    "(remember, these are the numbered labels of our Principal Components):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "importances=clf_Ca.feature_importances_\n",
+    "indices = np.argsort(importances)[::-1]\n",
+    "\n",
+    "#Print the feature ranking\n",
+    "print(\"Feature ranking:\")\n",
+    "\n",
+    "#for f in range(X.shape[1]):\n",
+    "for f in range(10):\n",
+    "    print(\"{:d}.  feature {:d} ({:f})\".format(f+1,indices[f],importances[indices[f]]))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For the most part, the important features appear to also correspond to the the PCs that explain the most variance.  \n",
+    "However, one shouldn't read too much into the feature importances - \n",
+    "a low rank does not necessarily mean that the feature isn't important in determining outputs, \n",
+    "because there may be high correlation between features.  \n",
```

Comment:
ok, made clear that the point I'm trying to make was for feature sets with high degree of correlation between features